classes/local/datasource/base.php

```diff
@@ -266,4 +302,24 @@ public function get_for_export(): ?\stdClass {
      */
     public function set_from_import(\stdClass $data) {
     }
+
+    /**
+     * Returns a hash of the content, representing the data stored for the datasource.
+     *
+     * @return string
+     */
+    public function get_content_hash(): string {
+        return '';
```

Comment:
thinking this might be better as abstract to force people to think about it

Reply:
done